File: ejercicios hechos/Java/estructuras repetitivas/act11.java
/*Programa: act11.java
 * 
 *Propósito: Suponiendo que hemos introducido una cadena por teclado que 
 * representa una frase (palabras separadas por espacios), realiza un programa 
 * que cuente cuantas palabras tiene.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 *
 *Diseño:
*/
import java.util.Scanner;
public class act11 {
    public static void main(String[] args) {
        System.out.println("Programa que representa una frase");
        System.out.println("---------------------------------");
         Scanner s = new Scanner(System.in);
        String cadena;
        int posicion = 0;
        int contadorPalabras = 0;

        System.out.print("Introduce una cadena: ");
        cadena = s.nextLine();

        while (cadena.charAt(posicion)==' ') {
          posicion++;
        }
        for (; posicion<=cadena.length()-1; posicion++) {
          if (cadena.charAt(posicion)==' ') {
            contadorPalabras++;
            while (posicion<=cadena.length()-1 && cadena.charAt(posicion)==' ') {
              posicion++;
            }
          }
        }
        if (cadena.charAt(cadena.length()-1)!=' ') {
          contadorPalabras++;
        }
        System.out.println("La frase tiene "+contadorPalabras+" palabras.");
    }
}

File: ejercicios terminados/Python/ejercicios repeticiones/act11.py
#Programa: act11.py
#Propósito: Suponiendo que hemos introducido una cadena por teclado que representa una frase (palabras separadas por espacios), realiza un programa que cuente cuantas palabras tiene.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 29/10/2019
#
# Análisis:
# Introduce el usuario una frase
# comprobamos cada posición de la cadena y 
# si hay espacio es por que hemos cambiado de palabra
 
# Diseño:
# Leemos cad
# para i en el rango la longitud de cad
#   si la posicion de la cadena es igual a espacio
#       se añade a un contador
# fin del bucle
# escribimos "el resultado es 'contador' palabras
# variables:  cad es la cadena, cont el contador, i el indice del for.

print("Contador de palabras")
print("---------------------\n")

# Leemos los datos
cad = str(input("Introduce una frase: "))
cont = 1

# Realizamos el contador
for i in range (len(cad)):
    if cad[i] == " ":
        cont = cont + 1

# Imprimimos el resutlado
print(f"Hay {cont} palabras en la frase")
